packages/wrangler/src/deploy/deploy.ts

```diff
@@ -189,26 +191,20 @@ export const validateRoutes = (routes: Route[], hasAssets: boolean) => {
 					`Paths are not allowed in Custom Domains`
 				);
 			}
-		} else if (hasAssets) {
+		} else if (assetsDir !== undefined) {
 			const pattern = typeof route === "string" ? route : route.pattern;
 			const components = pattern.split("/");
 
-			if (
-				// = ["route.com"]  bare domains are invalid as it would only match exactly that
-				components.length === 1 ||
-				// = ["route.com",""] as above
-				(components.length === 2 && components[1] === "")
-			) {
+			// If the route does not end with a wildcard, then error
+			// We may want to allow this in the future, but for now let's error
```

Comment:
Wondering what you think @GregBrimble @korinne - imo this is a totally valid use case but not sure we want to unleash it _just yet_.

Reply:
I'd personally just remove this restriction entirely. Otherwise, we'll have to remember to follow-up if we ever do infix splats in routes or whatever.